lib/pbench/server/cache_manager.py

```diff
@@ -844,7 +844,7 @@ def _get_metadata(tarball_path: Path) -> Optional[JSONOBJECT]:
         name = Dataset.stem(tarball_path)
         try:
             data = Tarball.extract(tarball_path, f"{name}/metadata.log")
-        except CacheExtractBadPath:
+        except Exception:
```

Comment:
This change doesn't seem like a step forward.  Previously, the function would return `None` only if the `metadata.log` file was not found, and it would raise an exception for other cases; now we're "eating" that exception, and I think we're losing information in the process.

This change does not, as far as I can tell, actually affect the caller, other than to increase the number of cases in which it receives `None`, so I strongly suggest reverting this change.

In fact, I would go one step further and just remove this `try` block -- let the caller deal with the problem.  (The caller is already, arguably, mishandling the `None` return, so raising an exception in all cases instead of returning `None` would likely be an improvement!)

Reply:
You're quibbling about logging visibility, really. The point was to be more forgiving and reject fewer datasets due to problems related to `metadata.log`. I decided not to add extra `logger.warning` messages to identify the details; but your comments make sense only if your intent was to indirectly argue that I should have.

Reply:
The way I read the existing code is that, in the event of failure, we either return `None` or we raise an exception; with your change, I don't think we raise exceptions any more -- we always return `None`.  That's adequate, but it doesn't provide the caller with much information (and, in turn, we as developers don't get much from the logging).

I don't see how this change makes the code more forgiving or causes it to reject fewer datasets.  What am I missing?

(I'm OK with minimizing the changes...but, toward that end, I think we can do that by reverting this change.  On the other hand, having better visibility would be nice, and I think the best way to accomplish that is to remove the `try` block altogether.  Thus, until you change my understanding, this fix seems to have negative value.)